CHANGELOG.md

```diff
@@ -1,3 +1,53 @@
+# v1.29.0 Release - 07/29/2021
+Note: This release comes with a new config version, `v2beta20`. To upgrade your skaffold.yaml, use `skaffold fix`. If you choose not to upgrade, skaffold will auto-upgrade as best as it can.
+
+Highlights:
+* Skaffold master -> main [#6263](https://github.com/GoogleContainerTools/skaffold/pull/6263)
+* Add state to event handler to track `logrus` output events [#6272](https://github.com/GoogleContainerTools/skaffold/pull/6272)
+* Remove skaffoldLogs endpoint, consolidate into events endpoint [#6296](https://github.com/GoogleContainerTools/skaffold/pull/6296)
+* fix: kubeContext override via flag [#6331](https://github.com/GoogleContainerTools/skaffold/pull/6331)
+* Added developer journey tutorial [#6201](https://github.com/GoogleContainerTools/skaffold/pull/6201)
+* Make skaffold reproducible [#6238](https://github.com/GoogleContainerTools/skaffold/pull/6238)
+
+Fixes:
+* Fix tail false but still log issue [#6299](https://github.com/GoogleContainerTools/skaffold/pull/6299)
+* skip manifest validation for default `kubectl deployer` [#6294](https://github.com/GoogleContainerTools/skaffold/pull/6294)
+* Fix switched container/pod names in app log event [#6215](https://github.com/GoogleContainerTools/skaffold/pull/6215)
+
+Updates and Refactors:
```

Comment:
I wonder if it's worth separating _updates_ from _refactors_

Reply:
Leaving as is for now, somewhat ambiguous to me at least the difference in some cases